lib/config/autoconfig.js

```diff
@@ -0,0 +1,367 @@
+
+        return failingRegistry;
+    },
+
+    /**
+     * Create an eslint config for any rules which only have one configuration
+     * in the registry.
+     *
+     * @returns {Object} An eslint config with rules section populated
+     */
+    createConfig: function() {
+        var ruleIds = Object.keys(this.rules),
+            config = {rules: {}};
+
+        ruleIds.forEach(function(ruleId) {
+            if (this.rules[ruleId].length === 1) {
+                config.rules[ruleId] = this.rules[ruleId][0].config;
+            }
+        }.bind(this));
+
+        return config;
+    },
+
+    /**
+     * Return a cloned registry containing only configs with a desired specificity
+     *
+     * @param   {number} specificity Only keep configs with this specificity
+     * @returns {Registry}           A registry of rules
+     */
+    filterBySpecificity: function(specificity) {
+        var ruleIds = Object.keys(this.rules),
+            newRegistry = new Registry();
+
+        newRegistry.rules = assign({}, this.rules);
+        ruleIds.forEach(function(ruleId) {
+            newRegistry.rules[ruleId] = this.rules[ruleId].filter(function(registryItem) {
+                return (registryItem.specificity === specificity);
+            });
+        }.bind(this));
+
+        return newRegistry;
+    },
+
+    /**
+     * Lint SourceCodes against all configurations in the registry, and record results
+     *
+     * @param   {Object[]} sourceCodes  SourceCode objects for each filename
+     * @param   {Object}   config       ESLint config object
+     * @param   {progressCallback} [cb] Optional callback for reporting execution status
+     * @returns {Registry}              New registry with errorCount populated
+     */
+    lintSourceCode: function(sourceCodes, config, cb) {
+        var totalFilesLinting,
+            lintConfig,
+            lintResults,
+            ruleSets,
+            ruleSetIdx,
+            filenames,
+            lintedRegistry;
+
+        lintedRegistry = new Registry();
+        lintedRegistry.rules = assign({}, this.rules);
+        ruleSets = lintedRegistry.buildRuleSets();
+        lintedRegistry = lintedRegistry.stripExtraConfigs();
+
+        debug("Linting with all possible rule combinations");
+        filenames = Object.keys(sourceCodes);
+        totalFilesLinting = filenames.length * ruleSets.length;
+        filenames.forEach(function(filename) {
+            debug("Linting file: " + filename);
+            ruleSetIdx = 0;
+            ruleSets.forEach(function(ruleSet) {
+                lintConfig = assign({}, config, {rules: ruleSet});
+                try {
+                    lintResults = eslint.verify(sourceCodes[filename], lintConfig);
+                    lintResults.forEach(function(result) {
+                        lintedRegistry.rules[result.ruleId][ruleSetIdx].errorCount += 1;
+                    });
+                    // Deallocate for GC
+                    lintResults = null;
```

Comment:
I would scope the variable to this function rather than defining it on the parent function.
That way its automatically GC'ed (if i am not wrong :smile: )